api/enki.py

```diff
@@ -223,7 +223,7 @@ def recent_activity(self, since):
         for element in data['result']['recentactivity']:
             identifier = IdentifierData(Identifier.ENKI_ID, element['id'])
             yield parser.extract_circulation(
-                identifier, element['availability']
+                identifier, element['availability'], {} #The recent activity API does not include format info
```

Comment:
`availability` is supposed to be a dictionary, so if the value is missing then it's appropriate to substitute an empty dictionary (as on line 611, below). But `formattype` is supposed to be a scalar value, so if the value is missing then the best substitute is `None`.